User Navigation

hsmitha26 opened the pull request:
Adds tests for registered user navigation.
Adds merchants_controller.
Adds Login form in sessions view.
Adds logout path and corresponding CRUD function in sessions controller.
CRUD function for login added in sessions controller.
All tests are passing.

hsmitha26 commented:
- Worked on getting all tests to pass.
- application.html.erb now has a new condition to control for existence of current user.

jalena-penaligon commented on spec/features/users/navigation_spec.rb:
```diff
@@ -25,9 +25,17 @@
-    xit 'I see the same links as a visitor' do
-      user = User.create(email: "j@gmail.com", password:"password", role:0)
-      visit root_path
+    it 'I see the same links as a visitor' do
```
I wonder if we can rephrase this slightly, since they see the same 1st 4 links, but then 2 links that are unique to a registered user. I think it may also be beneficial to add a test that we expect(page).to_not have_content for Login or Register?

hsmitha26 replied:
You are correct, again :)  I will change the wording to be more clear about what we are testing.

hsmitha26 commented on app/controllers/sessions_controller.rb:
```diff
@@ -1,4 +1,19 @@
+  end
+
+  def destroy
+    session[:user_id] = nil
```
You are correct!  I will change the line to session.clear.  Thank you for catching that.

hsmitha26 commented:
Made the following changes:
- adds edge case tests for registered user navigation.
- refactored feature test for navigation.
- changed code to session.clear in sessions controller for logout action

erin-king commented:
Looks good!
